### PR TITLE
Fit Viewport cleanup

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Changes how Fit Viewport works slightly.
From what I understand, the verb was created with the goal of
eliminating the pixel hunting required to get rid of letterboxing.

This works fine for stretch to fit, but ever since the command bar got
nuked it's ended up creating a lot of blackspace for targeted zoom
modes.

I've changed how it decides on its optimal width slightly, if the client
has a non stretch to fit zoom mode, we use it, the world icon size and
the clients view size to figure out the exact width we want.

The bars on the left and right have been bugging me forever. Want them
gone.

In light of this, I'm also making changing your zoom amount attempt to
fit the viewport, if you have the pref enabled.

Oh and I'm trying something with auto fit viewport stuff.
It currently waits a second between view change and fitting. The comment
implies this is to avoid winget strangeness from Login calls. When I
blamed it, I found a commit from antruk talking about client dropping,
and this being a potential fix.

Unfortunately none wrote down what dropping means, and anturk's
forgotten.

I'm making the assumption that it's related to attempting the related
winsets before an inital login is over.
I might be wrong about this, if I am we'll know what went wrong I
suppose.

## Why It's Good For The Game

Looks nicer

## Changelog

:cl:
fix: Fit viewport will properly well, fit your viewport now. Broke when we removed the status bar.
 It'll also attempt to autofit if you change your pixel scaling. Godspeed
/:cl:

### CODE REVIEW

File: code/datums/view.dm [hunk 1]
```diff
@@ -24,9 +25,12 @@
 
 /datum/view_data/proc/assertFormat()//T-Pose
 	winset(chief, "mapwindow.map", "zoom=0")
+	zoom = 0
 
 /datum/view_data/proc/resetFormat()//Cuck
-	winset(chief, "mapwindow.map", "zoom=[chief.prefs.read_preference(/datum/preference/numeric/pixel_size)]")
+	zoom = chief.prefs.read_preference(/datum/preference/numeric/pixel_size)
```
Comment (MrMelbert) [inline code/datums/view.dm:31]: can `chief` be null, as it's a client? 
or does view data ensure it's always there

File: code/datums/view.dm [hunk 2]
```diff
@@ -4,6 +4,7 @@
 	var/width = 0
 	var/height = 0
 	var/default = ""
+	var/zoom = 0
```
Comment (MrMelbert) [inline code/datums/view.dm:7]: auto-doc please, milord

Comment (LemonInTheDark) [reply code/datums/view.dm:7]: s my shitcode, I'll fix it as I like (sure)

File: code/datums/view.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/view.dm:31]: now that I think about it, I think the chief var here hangs references.
Will fix it to properly cleanup. Rule should be if the client doesn't exist, you'll have no way to call any of these procs
(That said something is really cursed in how byond handles clients as I'm sure you know. Maybe I should just use ?.)

Comment (MrMelbert) [reply code/datums/view.dm:31]: I don't know, any instance of `client.prefs` scares me for that reason. If you think it's fine though then it's w/e

Comment (LemonInTheDark) [reply code/datums/view.dm:31]: nah it shouldn't be like that you're right

**Pull request merged**